chameleon/os_login.py

```diff
@@ -47,14 +47,10 @@ def get_unscoped_token(request):
 def update_ks_password(request):
     try:
         logger.info('Synchronizing password to keystone for user: ' + request.POST.get('username'))
-        auth = v3.Password(auth_url=settings.OPENSTACK_KEYSTONE_URL,username=settings.OPENSTACK_SERVICE_USERNAME, password=settings.OPENSTACK_SERVICE_PASSWORD, \
-            project_id=settings.OPENSTACK_SERVICE_PROJECT_ID, project_name='services', user_domain_id="default")
-        sess = session.Session(auth=auth, timeout=5)
-        sess = adapter.Adapter(sess, interface='public', region_name=settings.OPENSTACK_TACC_REGION)
-        ks = ks_client.Client(session=sess, interface='public', region_name=settings.OPENSTACK_TACC_REGION)
-        user = filter(lambda this: this.name==request.POST.get('username'), ks.users.list())
+        ks_admin = get_admin_ks_client(request=request)
+        user = filter(lambda this: this.name==request.POST.get('username'), ks_admin.users.list())
```

Comment:
This looks cool, but it's expensive (I kno, I kno, it was already there, it was a carry-over from the sync script code ), with `ks_user = ks_admin.users.list(name=username)`  we can let keystone query the db for a specific user instead of asking it so get and return all users only for us to iterate through a list to get the one we need

Reply:
Cool, it is probably possible to do a direct lookup via username, I will check.

Reply:
Should be able to do it with, `ks_user = ks_admin.users.list(name=username)`